File: src/com/polaris/engine/options/Monitor.java
/**
 *
 */
package com.polaris.engine.options;

import org.lwjgl.glfw.GLFWGammaRamp;
import org.lwjgl.glfw.GLFWVidMode;

import static org.lwjgl.glfw.GLFW.*;

/**
 * @author lec50
 */
public class Monitor
{
	
	public static Monitor createMonitor(long monitor)
	{
		String name;
		int[] physicalSizeX = new int[1];
		int[] physicalSizeY = new int[1];
		int[] xpos = new int[1];
		int[] ypos = new int[1];
		GLFWVidMode videoMode;
		GLFWGammaRamp gammaRamp;
		
		name = glfwGetMonitorName(monitor);
		glfwGetMonitorPhysicalSize(monitor, physicalSizeX, physicalSizeY);
		glfwGetMonitorPos(monitor, xpos, ypos);
		videoMode = glfwGetVideoMode(monitor);
		gammaRamp = glfwGetGammaRamp(monitor);
		
		return new Monitor(name, monitor, physicalSizeX[0], physicalSizeY[0], xpos[0], ypos[0], videoMode, gammaRamp);
	}
	
	private final String monitorName;
	private final long monitorInstance;
	private final int physicalSizeX;
	private final int physicalSizeY;
	private final int monitorPosX;
	private final int monitorPosY;
	private final GLFWVidMode videoMode;
	private final GLFWGammaRamp gammaRamp;
	
	private Monitor(String name, long instance, int sizeX, int sizeY, int xpos, int ypos, GLFWVidMode mode, GLFWGammaRamp ramp)
	{
		monitorName = name;
		monitorInstance = instance;
		
		physicalSizeX = sizeX;
		physicalSizeY = sizeY;
		monitorPosX = xpos;
		monitorPosY = ypos;
		
		videoMode = mode;
		gammaRamp = ramp;
	}
	
	public String getName()
	{
		return monitorName;
	}
	
	public long getInstance()
	{
		return monitorInstance;
	}
	
	public int getSizeX()
	{
		return physicalSizeX;
	}
	
	public int getSizeY()
	{
		return physicalSizeY;
	}
	
	public int getPosX()
	{
		return monitorPosX;
	}
	
	public int getPosY()
	{
		return monitorPosY;
	}
	
	public GLFWVidMode getVideoMode()
	{
		return videoMode;
	}
	
	public GLFWGammaRamp getGammaRamp()
	{
		return gammaRamp;
	}
	
}

File: src/polaris/okapi/options/Monitor.kt
package polaris.okapi.options

import org.lwjgl.glfw.GLFW.*
import org.lwjgl.glfw.GLFWGammaRamp
import org.lwjgl.glfw.GLFWVidMode

/**
 * Created by Killian Le Clainche on 12/11/2017.
 */

fun createMonitor(monitorInstance : Long) : Monitor? {
    val name: String = glfwGetMonitorName(monitorInstance) ?: return null
    val physicalSizeX = IntArray(1)
    val physicalSizeY = IntArray(1)
    val xpos = IntArray(1)
    val ypos = IntArray(1)
    val videoMode: GLFWVidMode = glfwGetVideoMode(monitorInstance) ?: return null
    val gammaRamp: GLFWGammaRamp = glfwGetGammaRamp(monitorInstance) ?: return null

    glfwGetMonitorPhysicalSize(monitorInstance, physicalSizeX, physicalSizeY)
    glfwGetMonitorPos(monitorInstance, xpos, ypos)

    return Monitor(name, monitorInstance, physicalSizeX[0], physicalSizeY[0], xpos[0], ypos[0], videoMode, gammaRamp)
}

data class Monitor (val name: String, val instance: Long, val sizeX: Int, val sizeY: Int, val posX: Int, val posY: Int, val videoMode: GLFWVidMode, val gammaRamp: GLFWGammaRamp)